language-server/modules/langserver-compiler/src/main/java/org/ballerinalang/langserver/compiler/LSCompilerUtil.java

```diff
@@ -242,6 +242,37 @@ public static String getSourceRoot(Path filePath) {
         return fileRoot != null ? fileRoot : parentPath.toString();
     }
 
+    /**
+     * Returns top-level module path of a given file path.
+     *
+     * @param filePath file path
+     * @return top-level module path
+     */
+    public static Path getCurrentModulePath(Path filePath) {
+        Path projectRoot = Paths.get(LSCompilerUtil.getSourceRoot(filePath));
+        Path currentModulePath = projectRoot;
+        Path prevSourceRoot = filePath.getParent();
+        try {
+            if (prevSourceRoot == null || Files.isSameFile(prevSourceRoot, projectRoot)) {
+                return currentModulePath;
+            }
+            while (true) {
+                Path newSourceRoot = prevSourceRoot.getParent();
+                currentModulePath = prevSourceRoot;
+                String newSourceRootStr = (newSourceRoot != null) ? newSourceRoot.toString() : "";
+                if (newSourceRootStr.isEmpty() || "/".equals(newSourceRootStr) || newSourceRoot == null ||
+                        Files.isSameFile(newSourceRoot, projectRoot)) {
+                    // We have reached the project root
+                    break;
+                }
+                prevSourceRoot = newSourceRoot;
+            }
+        } catch (IOException e) {
+            // do nothing
```

Comment:
If the exception occurs is there a possibility to return an invalid module root?

Reply:
If we encounter I/O error, we are returning the `projectRoot` as `currentModulePath`. Logging error is unnecessary here IMO.

Reply:
No we don't need to log the error. I was wondering if an exception happens doesn't it mean that we should return something identifiable as invalid?

Reply:
Agree. Since throwing exception requires many changes to its usages of `getBLangPackage()` Let's keep fallback into `projectRoot`. I'll send a separate PR for it.